agent/xds/clusters.go

```diff
@@ -101,9 +103,24 @@ func makeUpstreamCluster(upstream structs.Upstream, cfgSnap *proxycfg.ConfigSnap
 	}
 
 	if c == nil {
+		conTimeout := 5 * time.Second
+		if toRaw, ok := upstream.Config["connect_timeout_ms"]; ok {
+			switch v := toRaw.(type) {
```

Comment:
The body of this `if` block feels like it would be better if it were extracted into a `parseTimeMillis` helper function or something like that, since it's not terribly envoy specific.